tools/export/nb/__init__.py

```diff
@@ -0,0 +1,402 @@
+            'c_flags': flags['c_flags'],
+            'cxx_flags': flags['cxx_flags'],
+            'ld_flags': self.flags['ld_flags'],
+            'asm_flags': self.flags['asm_flags'],
+            'common_flags': self.flags['common_flags'],
+            'target': self.target,
+            'include_paths': include_paths,
+            'forced_includes': forced_includes,
+            'c_sources': c_sources,
+            'cpp_sources': cpp_sources,
+            's_sources': s_sources,
+            'headers': headers,
+            'headers_folder': self.get_netbeans_file_list(sorted(headers)),
+            'sources_folder': self.get_netbeans_file_list(sorted(sources)),
+            'options': self.options,
+            'c_std': self.get_netbeans_c_std(c_std),
+            'cpp_std': self.get_netbeans_cpp_std(cpp_std),
+            'linker_script': self.ld_script,
+            'linker_libs': sys_libs,
+            'pp_cmd': preproc
+        }
+        return jinja_ctx
+
+    def generate(self):
+        """Generate Makefile, configurations.xml & project.xml Netbeans project file
+        """
+        # super(Netbeans, self).generate()
+
+        jinja_ctx = self.create_jinja_ctx()
+
+        print
+        print 'Create a GNU ARM Netbeans C++ managed project'
+        print 'Project name: {0}'.format(self.project_name)
+        print 'Target: {0}'.format(self.toolchain.target.name)
+        print 'Toolchain: {0}'.format(self.TOOLCHAIN)
+
+        if not exists(join(self.export_dir, 'nbproject')):
+            makedirs(join(self.export_dir, 'nbproject'))
+
+        self.gen_file('nb/configurations.tmpl', jinja_ctx, 'nbproject/configurations.xml')
+        self.gen_file('nb/project.tmpl', jinja_ctx, 'nbproject/project.xml')
+        self.gen_file('nb/mbedignore.tmpl', jinja_ctx, '.mbedignore')
+        self.gen_file('nb/Makefile.tmpl', jinja_ctx, 'Makefile')
+
+        print
+        print 'Done. Import the \'{0}\' project in Netbeans.'.format(self.project_name)
+
+    # -------------------------------------------------------------------------
+
+    @staticmethod
+    def filter_dot(str_in):
+        """
+        Remove the './' prefix, if present.
+        This function assumes that resources.win_to_unix()
+        replaced all windows backslashes with slashes.
+        """
+        if str_in is None:
+            return None
+        if str_in[:2] == './':
+            return str_in[2:]
+        return str_in
+
+    # -------------------------------------------------------------------------
+
+    @staticmethod
+    def get_all_profiles():
+        tools_path = dirname(dirname(dirname(__file__)))
+        file_names = [join(tools_path, "profiles", fn) for fn in os.listdir(
+            join(tools_path, "profiles")) if fn.endswith(".json")]
+
+        # print file_names
+
+        # profile_names = [basename(fn).replace(".json", "")
+        #                 for fn in file_names]
+        # print profile_names
```

Comment:
Could you remove commented out code? (5 lines)

Reply:
Done!